hscontrol/types/node.go

```diff
@@ -196,55 +201,41 @@ func (node *Node) HasIP(i netip.Addr) bool {
 	return false
 }
 
-// IsTagged reports if a device is tagged
-// and therefore should not be treated as a
-// user owned device.
-// Currently, this function only handles tags set
-// via CLI ("forced tags" and preauthkeys).
+// IsTagged reports if a device is tagged and therefore should not be treated
+// as a user-owned device.
+// When a node has tags, the tags define its identity (not the user).
 func (node *Node) IsTagged() bool {
-	if len(node.ForcedTags) > 0 {
+	// Check both node.Tags and AuthKey.Tags for backwards compatibility
+	// During migration, some nodes might have tags on PreAuthKey but not yet on node
+	if len(node.Tags) > 0 {
 		return true
 	}
 
 	if node.AuthKey != nil && len(node.AuthKey.Tags) > 0 {
 		return true
 	}
```

Comment:
This should not be the case anymore. Tags are added to Tags on node at registration time, and the authkey should not be relevant again after that point. So we need to remove this dependencies.